File: GideonLee/Assignments/bubble_sort.js
// Bubble Sort

var array = [1, 8, 5, 3, 7, 6, 2, 4];

function bubble_sort(arr) {
	for (var i = 0; i < arr.length; i++) {	
		for (var j = 0; j < arr.length-1; j++) {
			if (arr[j] > arr[j+1]) {
				var temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp; 
			}	
		}
	}
	return arr;
};

console.log(bubble_sort(array));

File: GideonLee/Assignments/python/bubble_sort.py
# Bubble Sort
from random import randint
import time

start_time = time.clock()

def createRandomList(): 
	arr = []
	for i in range(0, 100): 
		score = randint(0, 10000)
		arr.append(score)
	return arr
 
test = createRandomList()

def bubble_sort(arr):
	for outer in range(0, len(arr)):
		for inner in range(0, len(arr)):
			nextOne = inner + 1
			if nextOne < len(arr):
				if arr[inner] > arr[nextOne]:
					temp = arr[inner]
					arr[inner] = arr[nextOne]
					arr[nextOne] = temp
	return arr 

print bubble_sort(test)

print '\nThis took:', time.clock() - start_time, 'seconds'